src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -303,27 +315,23 @@ public static GenomeLocSortedSet loadIntervals(
         }
         // If it's a Feature-containing file, convert it to a list of intervals
         else if ( FeatureManager.isFeatureFile(IOUtils.getPath(arg)) ) {
-            rawIntervals.addAll(featureFileToIntervals(parser, arg));
-        }
-        // If it's an interval file, add its contents to the raw interval list
-        else if ( isIntervalFile(arg) ) {
             try {
-                rawIntervals.addAll(intervalFileToList(parser, arg));
-            }
-            catch ( final UserException.MalformedGenomeLoc e ) {
-                throw e;
-            }
-            catch ( final Exception e ) {
-                throw new UserException.MalformedFile(new File(arg), "Interval file could not be parsed in any supported format.", e);
+                rawIntervals.addAll(featureFileToIntervals(parser, arg));
+            } catch (final IllegalArgumentException e){
+                throw new UserException.MalformedFile(IOUtils.getPath(arg), "Failure while loading intervals from file.", e);
             }
         }
+        // If it's an interval file, add its contents to the raw interval list
+        else if ( isGatkIntervalFile(arg) ) {
+            rawIntervals.addAll(gatkIntervalFileToList(parser, arg));
+        }
         // If it's neither a Feature-containing file nor an interval file, but is an existing file, throw an error.
         // Note that since contigs can contain periods in their names, we can't use the mere presence of an "extension"
         // as evidence that the user intended the String to be interpreted as a file.
         else if ( new File(arg).exists() ) {
             throw new UserException.CouldNotReadInputFile(arg, String.format("The file %s exists, but does not contain Features " +
                     "(ie., is not in a supported Feature file format such as vcf, bcf, or bed), " +
-                    "and does not have one of the supported interval file extensions (" + INTERVAL_FILE_EXTENSIONS + "). " +
+                    "and does not have one of the supported interval file extensions (" + GATK_INTERVAL_FILE_EXTENSIONS + "). " +
```

Comment:
Kind of a nit, but this error message will no longer include ".interval_list", which is an acceptable extension.

Reply:
technically interval_list is a feature file now